packages/solid/src/reactive/signal.ts

```diff
@@ -141,39 +151,45 @@ export type SignalOptions<T> = { name?: string; equals?: false | ((prev: T, next
  *
  * @description https://www.solidjs.com/docs/latest/api#createsignal
  */
-export function createSignal<T>(): [get: Accessor<T | undefined>, set: Setter<T | undefined>];
-export function createSignal<T>(
-  value: T,
-  options?: { equals?: false | ((prev: T, next: T) => boolean); name?: string; internal?: boolean }
-): [get: Accessor<T>, set: Setter<T>];
-export function createSignal<T>(
-  value?: T,
-  options?: { equals?: false | ((prev: T, next: T) => boolean); name?: string; internal?: boolean }
-): [get: Accessor<T>, set: Setter<T>] {
+export function createSignal<T>(): Signal<T | undefined>;
+export function createSignal<T>(value: T, options?: SignalOptions<T>): Signal<T>;
+export function createSignal<T>(value?: T, options?: SignalOptions<T>): Signal<T> {
   options = options ? Object.assign({}, signalOptions, options) : signalOptions;
-  const s: Signal<T> = {
+
+  const s: SignalState<T> = {
     value,
     observers: null,
     observerSlots: null,
     pending: NOTPENDING,
     comparator: options.equals || undefined
   };
+
   if ("_SOLID_DEV_" && !options.internal)
     s.name = registerGraph(options.name || hashValue(value), s as { value: unknown });
 
   return [
     readSignal.bind(s),
-    ((value: T extends Function ? never : T | ((p?: T) => T)) => {
+    value => {
```

Comment:
Removing the type annotation here means that `value` is typed as `any`, which obviously results in an error when building the types.

Reply:
Yes indeed! On a walk. Will get it in just a bit.